src/server/transport.rs

```diff
@@ -55,16 +55,50 @@ pub trait RaftStoreRouter: Send + Clone {
     ) -> RaftStoreResult<()> {
         self.try_send(StoreMsg::new_batch_raft_snapshot_cmd(batch, on_finished))
     }
+
+    // Send significant message. We should guarantee that the message can't be dropped.
+    fn significant_send(&self, msg: StoreMsg) -> RaftStoreResult<()>;
+
+    // Report the peer of the region is unreachable.
+    fn report_unreachable(&self, region_id: u64, to_peer_id: u64) -> RaftStoreResult<()> {
+        self.significant_send(StoreMsg::Unreachable {
+            region_id,
+            to_peer_id,
+        })
+    }
+
+    // Report the sending snapshot status to the peer of the region.
+    fn report_snapshot_status(
+        &self,
+        region_id: u64,
+        to_peer_id: u64,
+        status: SnapshotStatus,
+    ) -> RaftStoreResult<()> {
+        self.significant_send(StoreMsg::SnapshotStatus {
+            region_id: region_id,
+            to_peer_id: to_peer_id,
+            status: status,
+        })
+    }
 }
 
 #[derive(Clone)]
 pub struct ServerRaftStoreRouter {
     pub ch: SendCh<StoreMsg>,
+    // Some tests need Sync for ServerRaftStoreRouter, so here we use Arc + Mutex here.
+    // TODO: refactor the test and remove the limitation later.
+    pub significant_msg_sender: Arc<Mutex<Sender<StoreMsg>>>,
```

Comment:
How about `SyncSender`?

Reply:
SyncSender may send failed with Full error, but here we must always send successfully unless shutdown.